libs/core/langchain_core/runnables/history.py

```diff
@@ -364,10 +368,19 @@ def get_input_schema(
             return super_schema
 
     def _get_input_messages(
-        self, input_val: Union[str, BaseMessage, Sequence[BaseMessage]]
+        self, input_val: Union[str, BaseMessage, Sequence[BaseMessage], dict]
```

Comment:
you can use TypedDict here if you care -- it's extra code, but more precise in terms of what's expected

from typing_extensions import TypedDict

```
class InputDict(TypedDict):
  input: Sequence[BaseMessage] ? 
```